File: FirstApp/Stock/UserControl/SuggestionControl.vb
﻿Public Class SuggestionControl
    Private _Attribute As Item.Item_Attribute

    Private _currentAttribute As Item.Item_Attribute

    Public Property CurrentAttribute As Item.Item_Attribute
        Get
            Return _currentAttribute
        End Get
        Set(value As Item.Item_Attribute)
            _currentAttribute = value
            If value Is Nothing Then Exit Property
            With _currentAttribute
                AttrIconBT.Visible = True
                Label2.Visible = True
                AttrIconBT.Text = $"{{{ .DataType}{If(.IsCurrency, "-₹", "")}}} { .Name}" + If(.IsOptional, "*", "")
                If .IsConstant Then
                    AttrIconBT.Font = New Font(AttrIconBT.Font, FontStyle.Underline)
                End If
                Size = New Point(Size.Width + TextRenderer.MeasureText(AttrIconBT.Text, AttrIconBT.Font).Width + 2, Height)
                If .IsFormula Then
                    DefValueLB.Text = "{RULE}"
                ElseIf .DataType = Item.DataType.List_Type Then
                    DefValueLB.Text = .DefaultValue.Split("`")(0)
                Else
                    DefValueLB.Text = .DefaultValue
                End If

                AttrIconBT.IconChar = .Icon
            End With

        End Set
    End Property
End Class

File: FirstApp/Stock/UserControl/SuggestionControl.cs
using System.Drawing;
using System.Windows.Forms;

namespace FirstApp
{
    public partial class SuggestionControl
    {
        private Item.Item_Attribute _Attribute;

        private Item.Item_Attribute _currentAttribute;

        public Item.Item_Attribute CurrentAttribute
        {
            get
            {
                return _currentAttribute;
            }
            set
            {
                _currentAttribute = value;
                if (value is null)
                    return;
                {
                    ref var withBlock = ref _currentAttribute;
                    AttrIconBT.Visible = true;
                    Label2.Visible = true;
                    AttrIconBT.Text = $"{{{withBlock.DataType}{(withBlock.IsCurrency ? "-₹" : "")}}} {withBlock.Name}" + (withBlock.IsOptional ? "*" : "");
                    if (withBlock.IsConstant)
                    {
                        AttrIconBT.Font = new Font(AttrIconBT.Font, FontStyle.Underline);
                    }
                    Size = (Size)new Point(Size.Width + TextRenderer.MeasureText(AttrIconBT.Text, AttrIconBT.Font).Width + 2, Height);
                    if (withBlock.IsFormula)
                    {
                        DefValueLB.Text = "{RULE}";
                    }
                    else if ((withBlock.DataType ?? "") == Item.DataType.List_Type)
                    {
                        DefValueLB.Text = withBlock.DefaultValue.Split('`')[0];
                    }
                    else
                    {
                        DefValueLB.Text = withBlock.DefaultValue;
                    }

                    AttrIconBT.IconChar = withBlock.Icon;
                }

            }
        }

        public SuggestionControl()
        {
            InitializeComponent();
        }
    }
}